### PR TITLE
Fix unsafe globals in Interval and Region3

### PR DESCRIPTION
**Pull request by hidmic**
# 🦟 Bug fix

Relates to #269.

## Summary
This PR fixes a potential global initialization order issue introduced in #388 and #390 by making all `Interval` and `Region3` construction forms `constexpr`.

## Checklist
- [x] Signed all commits for DCO
- [ ] Added tests
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [x] `codecheck` passed (See [contributing](https://ignitionrobotics.org/docs/all/contributing#contributing-code))
- [x] All tests passed (See [test coverage](https://ignitionrobotics.org/docs/all/contributing#test-coverage))

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

### CODE REVIEW

**Review by chapulina — APPROVED** [review]
LGTM, but I'd like to make sure @scpeters approves too before merging

**Comment by scpeters** [comment]
it seems right to me, but I think @azeey understands this issue better than me. I think the right way to test this is with asan using https://github.com/ignitionrobotics/ign-cmake/pull/210, but I haven't had a chance to do so. I think it's fine to merge without my approval. Once we get CI with sanitizers, we'll know for sure :)

**Comment by chapulina** [comment]
Sounds reasonable, we can revisit when the sanitizers are running in CI, if needed

**Pull request merged**
